amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -346,6 +346,16 @@ public static AmpActivityVersion initializeActivity(AmpActivityVersion act) thro
         return act;
     }
     
+    public static Map<Long, Map<String, List<CompareOutput>>> compareActivities(Long[] ActivitiesId) throws Exception {
+        
+        Session session = PersistenceManager.getCurrentSession();
+        Map<Long, Map<String, List<CompareOutput>>> activitiesCollection = new HashMap<Long, Map<String, List<CompareOutput>>>();
+        for (int i=0; i < ActivitiesId.length; i++) {
+            Map<String, List<CompareOutput>> ActivityList = compareActivities(ActivitiesId[i]);
```

Comment:
I have added an If condition to check for Null but it checks compareActivities(ActivitiesId[i]) method twice. I think this will degrade performance. shouldn't this check be on compareActivities(ActivitiesId[i]) method (AMPET-141)?

Reply:
yes it will degrade performance what you need to do is return null in the method that compares to it previous version if it does not find an activity, and then don't add it to the map if the comparison is null